netboot/main.go

```diff
@@ -225,6 +241,82 @@ func boot(ifname string, dhcp dhcpFunc) error {
 	return nil
 }
 
+func getScheme(urlstring string) (string, error) {
+	u, err := url.Parse(urlstring)
+	if err != nil {
+		return "", err
+	}
+	scheme := strings.ToLower(u.Scheme)
+	if scheme != "http" && scheme != "https" {
+		return "", fmt.Errorf("URL scheme '%s' must be http or https", scheme)
+	}
+	return scheme, nil
+}
+
+func loadCaCerts() (*x509.CertPool, error){
+	rootCAs, err := x509.SystemCertPool()
+	if err != nil {
+		return nil, err
+	}
+	if rootCAs == nil {
+		debug("certs: rootCAs == nil")
+		rootCAs = x509.NewCertPool()
+	}
+	base64Certs, err := ioutil.ReadFile(*caCertFile)
+	if err != nil {
+		return nil, fmt.Errorf("could not find cert file '%v' - %v", *caCertFile, err)
+	}
+	// TODO: Decide if this should also support compressed certs
+	// Might be better to have a generic compressed config API
+	decLen := base64.StdEncoding.DecodedLen(len(base64Certs))
+	certs := make([]byte, decLen)
+	n, err := base64.StdEncoding.Decode(certs, base64Certs)
+	if err != nil {
+		return nil, fmt.Errorf("failed to append %s to RootCAs: %v", *caCertFile, err)
+	}
+	if ok := rootCAs.AppendCertsFromPEM(certs); !ok {
+		debug("No certs extracted from VPD, using system certs only")
+	} else {
+		debug("%d bytes of certs extrated from VPD", n)
+	}
+	return rootCAs, nil
+
+}
+
+func getClientForBootfile(bootfile string) (*http.Client, error) {
+	var client *http.Client
+	scheme, err := getScheme(bootfile)
+	if err != nil {
+		return nil, err
+	}
+
+	if scheme == "https" {
+		var config *tls.Config
+		if *skipCertVerify {
+			config = &tls.Config{
+				InsecureSkipVerify: true,
+			}
+		} else if *caCertFile != "" {
+			rootCAs, err := loadCaCerts()
+			if err != nil {
+				return nil, err
+			}
+			config = &tls.Config{
+				RootCAs: rootCAs,
+			}
+		}
+		tr := &http.Transport{TLSClientConfig: config}
+		client = &http.Client{Transport: tr}
+		debug("https client setup to use certs from VPD")
```

Comment:
should this debug line also say whether we are skipping certificate verification?